### PR TITLE
Adding FTU

### LINKED ISSUE #583: Implement Discovery view FTU
In an effort to reduce frustration when landing on the Discovery View, we are introducing a Discovery View FTU.
#### Notes & Requirements
- FTU should display once only, after used has created their account.
- Localization of this message is key. Let's talk about how we can achieve this. Should we load two new fonts (Sharpie font for English, Portuguese, Indonesian && Google Noto for Bengali)? Or should we make these modals images and swap them based on locale?
##### Text Strings

English: Explore stories around the world ... or make your own!
Brazilian Portuguese: Explore histórias ao redor do mundo ... ou faça a sua !
Indonesian: Temukan cerita dari seluruh dunia ... atau buatlah ceritamu sendiri!
Bengali: আপনার চারপাশের দুনিয়ার গল্পগুলো আবিষ্কার করুন ... অথবা নিজেই তৈরী করুন!
###### English

intro english 1x

---
###### Brazilian Portuguese

intro brazilian portuguese 1x

---
###### Indonesian

intro indonesian 1x

---
###### Bengali

intro bengali 1x

### LINKED ISSUE #630: Implement Play View Remix FTU
#### Parameters
- FTU will appear on the users' 2nd project lifetime view. I think the first Project view experience they have should just focus on the content, not on the UI. In their 2nd project view, they will have context of what viewing a project is about – we can introduce Remix then.
#### Design

play page remix ftu
##### Text String

`Make your own version of this - hit ‘Remix!’`
#### Asset

Asset available: https://drive.google.com/open?id=0ByIoeeW0a3R_bFJXa242RllFQ0U

### LINKED ISSUE #631: Implement Project Editor Share FTU
#### Parameters
- Modal will show up after user has created their 2nd page with content. My rationale is that one would be more likely to share a project that has more than 1 page.
#### Design

tile map view ftu
##### Text String

`Share this project with anyone!`
##### Asset

Asset available: https://drive.google.com/open?id=0ByIoeeW0a3R_bFJXa242RllFQ0U

### PR DESCRIPTION
**Pull request by gvn**
Closes #583 
Closes #631 
Closes #630

### CODE REVIEW

**Comment by gvn** [comment]
@k88hudson Do we have an existing JS exposed API to see if it's the user's first time opening the app, or is that something we need to add?

**Comment by gvn** [comment]
@ryanw-se Do you have any thoughts on adding a query method to the JS bridge API to see if it's a user's first time in the app? 

There are going to be some other variants for "first time use" as well, such as "users' 2nd project lifetime view" and "after user has created their 2nd page with content", but we can focus on first time opening the app first...

**Comment by ryanwarsaw** [comment]
@gvn Completely do-able, we could create a new method in the JS bridge to add a boolean to the SharedPreferences that we can check for, to see whether or not it is the users first time opening the app. The only downfall to this solution is, SharedPreferences are stored when the app is upgraded. Not sure if that is okay, or if we want the first time boolean to reset when you upgrade to the next version of the app.

**Comment by gvn** [comment]
I think it's actually ideal for the FTU boolean to persist over to an updated app. 

Honestly, it would probably be quite annoying to keep seeing FTU come back every time you update! :+1:

**Comment by gvn** [comment]
@ryanw-se Would you like me to make a ticket for you to work on for this?

**Comment by ryanwarsaw** [comment]
@gvn Sure, I got my hands tied with a couple other things so it may take me a couple days before I can get to it :)

**Comment by gvn** [comment]
Awesome, thanks! I just made a ticket for you – #642 :cake:

**Comment by gvn** [comment]
@ryanw-se I closed that ticket...I'm just going to use sharedPreferences as it already exists w. a ftu namespace.

File: src/components/ftu/ftu.jsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+var FTU = React.createClass({
+  mixins: [require('react-intl').IntlMixin],
+  propTypes: {
+    singleMessage: React.PropTypes.string,
+    rightMessage: React.PropTypes.string,
+    leftMessage: React.PropTypes.string
+  },
+  getInitialState: function () {
+    return {
+      isVisible: !this.props.hidden
```
Comment (k88hudson) [inline src/components/ftu/ftu.jsx]: Using state here creates a weird fragmented situation where if you change `this.state.isVisible` to false, whatever you were using to set `this.props.hidden` won't get updated. What do you think about passing in a `setVisibility` function or something instead?

**Comment by k88hudson** [comment]
Looks good, I would recommend not fragmenting the `visibility` state between the parent/child for FTU

**Comment by gvn** [comment]
@k88hudson I just got rid of that prop and made the FTU's API the only way to set its visibility. That should solve any fragmentation... :dancer:

**Pull request merged**